csrc/python_frontend/fusion_definition.h

```diff
@@ -45,6 +45,25 @@ struct Tensor {
     return index;
   }
 
+  bool operator==(const Tensor& other) const {
```

Comment:
In `FusionDefinition::tensors()`, we're creating new Tensors with the same state. Checking only Tensor addresses could create duplicates.

Reply:
> Checking only Tensor addresses could create duplicates.

that's not what I was suggesting though. It's just a short-cut if we have the same address then we don't need to check for each field. Otherwise, the existing logic will kick in to detect different Tensors pointing to the same TensorView/State.

But it's not a big issue.